contracts/cw-ics721-bridge/src/contract.rs

```diff
@@ -293,17 +304,21 @@ fn callback_create_vouchers(
 }
 
 fn callback_redeem_vouchers(
-    deps: Deps,
+    deps: DepsMut,
     receiver: String,
     redeem: VoucherRedemption,
 ) -> Result<Response, ContractError> {
     let VoucherRedemption { class, token_ids } = redeem;
-    let nft_contract = CLASS_ID_TO_NFT_CONTRACT.load(deps.storage, class.id)?;
+    let nft_contract = CLASS_ID_TO_NFT_CONTRACT.load(deps.storage, class.id.clone())?;
     let receiver = deps.api.addr_validate(&receiver)?;
     Ok(Response::default().add_messages(
         token_ids
             .into_iter()
             .map(|token_id| {
+                // token metadata is set in mint, on voucher creation, and removed here on voucher redemption
+                CLASS_TOKEN_ID_TO_TOKEN_METADATA
+                    .remove(deps.storage, (class.id.clone(), token_id.clone()));
```

Comment:
this would clear the token metadata on the source chain if a sink chain returned a debt voucher. it seems to me that we actually want to clear this on the sink chain. otherwise, sending a debt-voucher away and returning it would cause its metadata to be lost.

it seems to me that the appropriate place to do this is in `ibc_packet_ack` when we burn the token if it is returning to its source.